bazel/packaging/BUILD

```diff
@@ -0,0 +1,34 @@
+load("@rules_pkg//pkg:mappings.bzl", "pkg_files", "strip_prefix")
+load("@rules_pkg//pkg:tar.bzl", "pkg_tar")
+
+pkg_files(
+    name = "redpanda_binary_with_runfiles",
+    srcs = [
+        "//src/v/redpanda",
+    ],
+    # Including runfiles has the effect of copying all of the shared libraries needed to run the binary.
+    # Although they always retain the directory structure that is found in bazel build output.
```

Comment:
>  Although they always retain the directory structure that is found in bazel build output.

is this an important point, or just an fyi?

Reply:
Its important because it is why we need to flatten the output

Reply:
I added more comments, does that help clarify?

Reply:
yeh thanks!